src/big/big_float.cr

```diff
@@ -82,6 +82,14 @@ struct BigFloat < Float
     new(mpf)
   end
 
+  # :nodoc:
+  def initialize(*, str : String, base : Int, precision : Int)
+    LibGMP.mpf_init2(out @mpf, precision.to_u64)
+    if LibGMP.mpf_set_str(self, str, base) == -1
+      raise ArgumentError.new("Invalid BigFloat: #{str.inspect}")
+    end
+  end
```

Comment:
I'm not sure if it's a good idea to add an entire constructor just for testing purposes. Even if it's nodoc. Maybe this could be implemented directly in the spec helper instead? Or is there any other use case?